src/components/Api.js

```diff
@@ -0,0 +1,92 @@
+export default class Api {
+  constructor(data) {
+    this._url = data.baseUrl;
+    this._headers = data.headers;
+    this._avatar = data.avatar;
+  }
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  async getInitialCards() {
+    const res = await fetch(
+      "https://around-api.en.tripleten-services.com/v1/cards",
+      {
+        headers: this._headers,
+      }
+    );
+    return this._checkResponse(res);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.